nautobot/docs/development/apps/api/ui-extensions/object-views.md

```diff
@@ -25,10 +34,85 @@ When a TemplateExtension is instantiated, context data is assigned to `self.cont
 
 For example, accessing `{{ request.user }}` within a template will return the current user.
 
-Declared subclasses should be gathered into a list or tuple for integration with Nautobot. By default, Nautobot looks for an iterable named `template_extensions` within a `template_content.py` file. (This can be overridden by setting `template_extensions` to a custom value on the app's `NautobotAppConfig`.) An example is [below](#example-adding-object-details-and-tabs).
+## Adding Detail Panels
+
+### Via `object_detail_panels`
+
++++ 2.4.0
+
+The `TemplateExtension.object_detail_panels` should be a list or tuple of Panel objects (as provided by the `nautobot.apps.ui` module). A variety of base classes are available; refer to the relevant section of documentation for details.
+
+For example:
+
+```python title="example_app/template_content.py"
+from nautobot.apps.ui import ObjectTextPanel, SectionChoices, TemplateExtension
+
+
+class CircuitContent(TemplateExtension):
+    model = "circuits.circuit"
+
+    object_detail_panels = (
+        ObjectTextPanel(
+            weight=100,
+            label="Example App Text Panel",
+            section=SectionChoices.LEFT_HALF,
+            render_as=ObjectTextPanel.RenderOptions.CODE,
+            object_field="description",
+        ),
+    )
+
+template_extensions = [CircuitContent]
+```
+
+### Via the `*_page()` Methods (Deprecated)
+
+The `left_page()`, `right_page()`, and `full_width_page()` methods each simply return a fragment of HTML. You are responsible for ensuring that the returned HTML is properly constructed and doesn't break the page layout and rendering.
 
 ## Adding Detail Tabs
 
+### Via `object_detail_tabs`
+
++++ 2.4.0
+
+The `TemplateExtension.object_detail_tabs` should be a list or tuple of Tab objects (as provided by the `nautobot.apps.ui` module). Two base classes are available:
+
+* `Tab` - add a tab and its contents to the main object detail page, rendered inline with the rest of that page. Best used for quick-rendering content.
+* `DistinctViewTab` - add a tab to the main object detail page that links to a distinct view of its own when clicked. Best used for more involved content.
```

Comment:
I am a little confused as to what a distinct view is. Is it a view that requires a page reload or a different url?

Reply:
Either/both, yes.

Reply:
Perhaps RedirectTabView?

Reply:
To me that implies that it's got an HTTP Redirect involved, whereas normally it's just going to be a hyperlink directly to the relevant URL with no redirect involved.